scripts/backfill-inbox-message-contact-ids.ts

```diff
@@ -0,0 +1,163 @@
+          })
+          .from(ContactsTable)
+          .where(
+            or(
+              eq(ContactsTable.email, message.source),
+              eq(ContactsTable.slackId, message.source)
+            )
+          );
+
+        if (matchingContacts.length === 1) {
+          const contact = matchingContacts[0];
+
+          if (!this.isPreviewMode) {
+            await db
+              .update(InboxMessagesTable)
+              .set({ contactId: contact.id })
+              .where(eq(InboxMessagesTable.id, message.id));
+          }
+
+          this.logSuccess(
+            `${
+              this.isPreviewMode ? "[PREVIEW] Would update" : "Updated"
+            } message ${message.id} with contact ${contact.id} (source: ${
+              message.source
+            })`
+          );
+          matchedCount++;
+        } else if (matchingContacts.length > 1) {
+          this.logWarning(
+            `Multiple contacts found for source "${message.source}" - skipping message ${message.id}`
+          );
+          errorCount++;
+        } else {
+          if (!this.isPreviewMode) {
+            await db
+              .delete(InboxMessageOperationsTable)
+              .where(
+                eq(InboxMessageOperationsTable.inboxMessageId, message.id)
+              );
+
+            await db
+              .delete(OutboxMessagesTable)
+              .where(eq(OutboxMessagesTable.parentInboxMessageId, message.id));
+
+            await db
+              .delete(InboxMessagesTable)
+              .where(eq(InboxMessagesTable.id, message.id));
+          }
+
+          this.logWarning(
+            `${
+              this.isPreviewMode ? "[PREVIEW] Would delete" : "Deleted"
+            } message ${message.id} - no matching contact for source: ${
+              message.source
+            }`
+          );
+          deletedCount++;
+        }
+      } catch (error) {
+        this.logError(`Error processing message ${message.id}: ${error}`);
+        errorCount++;
+      }
+    }
+
+    this.logSection("Migration Summary");
+    this.logSuccess(
+      `Total messages processed: ${messagesWithoutContactId.length}`
+    );
+    this.logSuccess(`Messages matched and updated: ${matchedCount}`);
+    this.logWarning(`Messages deleted (no contact found): ${deletedCount}`);
+    if (errorCount > 0) {
+      this.logError(`Messages with errors: ${errorCount}`);
+    }
+
+    if (this.isPreviewMode) {
+      const previewSummary = `
+## Contact ID Backfill Migration Preview
+
+**Summary:**
+- Total messages without contact IDs: ${messagesWithoutContactId.length}
+- Messages that would be updated: ${matchedCount}
+- Messages that would be deleted: ${deletedCount}
+- Messages with errors/multiple matches: ${errorCount}
+
+**Actions:**
+- ✅ Update messages where source matches exactly one contact (email or slackId)
+- ⚠️ Delete messages where no matching contact is found
+- ⚠️ Skip messages with multiple potential contact matches
```

Comment:
If multiple contacts are found, just arbitrarily pick one, maybe the first one that was created